source/analysis/AM2DSummingAB.cpp

```diff
@@ -315,40 +317,92 @@ bool AM2DSummingAB::axisValues(int axisNumber, int startIndex, int endIndex, dou
 
 	int otherAxis = (sumAxis_ == 0) ? 1 : 0;
 
-	if (startIndex >= inputSource_->axisInfoAt(otherAxis).size || endIndex >= inputSource_->axisInfoAt(otherAxis).size)
+	if (startIndex >= inputSource_->size(otherAxis) || endIndex >= inputSource_->size(otherAxis))
 		return false;
 
 	return inputSource_->axisValues(otherAxis, startIndex, endIndex, outputValues);
 }
 
+void AM2DSummingAB::setSumAxis(int sumAxis)
+{
+	if((unsigned)sumAxis >= 2)
```

Comment:
I believe this introduces a rare corner case if sumAxis is passed as INT_MIN or INT_MIN + 1.
If we were to take a 32 bit int, with INT_MIN = -4294967296. The cast would add 2^32 in order to produce a valid unsigned value, and produce 0 for INT_MIN or 1 for INT_MIN + 1, which would then fail this if block and drop through to the rest of the code.

Obviously the chances of this are very minimal, but it would probably be more straight forward to just go with:
```C++
if(sumAxis < 0 || sumAxis >=2)
    return;
```